### ISSUE TITLE
Compress option to gzip compress files.

### ISSUE THREAD

**Issue by codenut** [Vinelab/cdn#85]
Thank you for creating this library! Is it possible to add a compress option that will gzip compress the files that will be uploaded in S3. The use case for this is that s3 doesnt automatically compress  javascript,css,etc. assets and it would be nice this option can be added. 

`'compres' => ['css', 'js']`

**Comment by Mulkave** [comment]
@codenut do you mean to automatically gzip before uploading? And upload the compressed files? We're not talking about minification here though, right?

**Comment by codenut** [comment]
@Mulkave yes, that is correct. Just gzip the file before uploading without minification. And also set the ContentEncoding to gzip.

Theres an implementation of it on the repository that I forked.

https://github.com/codenut/cdn/commit/204ffbbb30f249fd437e6422e2fa9e05101b2daa

Thanks!

**Comment by Mulkave** [comment]
@codenut cool! Can you please open a PR of your implementation?

**Comment by uniring** [comment]
+1 for this feature, can someone take a look at @codenut implementation?
